### PR TITLE
Enabling OPC Events related unit tests.

### PR DESCRIPTION
**Pull request by karok2m**
Changes:
* Enabled OPC events related tests in `PublishedNodesJobConverterTests`.
* Removed comment about priorities of id generation that is no longer valid.
* Enabled tests in `ReversibleEncodingIntegrationTests` suite.
* Minor addition to comments.

### CODE REVIEW

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs [hunk 1]
```diff
@@ -2437,7 +2463,7 @@ private class StandaloneIdentity : IIdentity {
                Assert.Single(dataSetWriter.DataSet.DataSetSource.PublishedEvents.PublishedData));
 
             var eventModel = jobs[0].WriterGroup.DataSetWriters[0].DataSet.DataSetSource.PublishedEvents.PublishedData[0];
-            Assert.Equal("i=2253", eventModel.Id);
+            //Assert.Equal("i=2253", eventModel.Id);
```
Comment (karok2m) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs]: Add value and validate or remove check.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs [hunk 2]
```diff
@@ -2368,7 +2388,7 @@ private class StandaloneIdentity : IIdentity {
...
```
Comment (karok2m) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs]: Add value and validate or remove check.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs [hunk 3]
```diff
@@ -2222,19 +2236,19 @@ private class StandaloneIdentity : IIdentity {
             Assert.Empty(jobs[3].WriterGroup.DataSetWriters[0].DataSet.DataSetSource.PublishedEvents.PublishedData);
 
             var dataItemModel = jobs[0].WriterGroup.DataSetWriters[0].DataSet.DataSetSource.PublishedVariables.PublishedData[0];
-            Assert.Equal("i=2258", dataItemModel.Id);
+            //Assert.Equal("i=2258", dataItemModel.Id);
```
Comment (karok2m) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs]: Add value and validate or remove check.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs [hunk 4]
```diff
@@ -2076,6 +2078,13 @@ private class StandaloneIdentity : IIdentity {
 ");
             using var schemaReader = new StreamReader("Storage/publishednodesschema.json");
```
Comment (dacolgit) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs]: >using var schemaReader = new StreamReader("Storage/publishednodesschema.json"); [](http://example.com/codeflow?start=12&length=79)

I think it should be removed, like in the following tests

File: modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/ReversibleEncodingIntegrationTests.cs [hunk 5]
```diff
@@ -23,13 +23,15 @@ public class ReversibleEncodingIntegrationTests : PublisherIntegrationTestBase,
...
+        [Theory]
         [InlineData(@"./PublishedNodes/SimpleEvents.json")]
         public async Task CanEncodeWithoutReversibleEncodingTest(string publishedNodesFile) {
             // Arrange
             // Act
-            var result = await ProcessMessagesAsync(publishedNodesFile, arguments: new[] { "--mm=PubSub", "--UseReversibleEncoding=False" });
+            var result = await ProcessMessagesAsync(
+                publishedNodesFile,
+                new[] { "--mm=PubSub", "--UseReversibleEncoding=False" }
+            );
```
Comment (dacolgit) [inline modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/ReversibleEncodingIntegrationTests.cs:35]: add ConfigureAwait(false)?

File: modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/ReversibleEncodingIntegrationTests.cs [hunk 6]
```diff
@@ -55,13 +57,15 @@ public class ReversibleEncodingIntegrationTests : PublisherIntegrationTestBase,
...
+        [Theory]
         [InlineData(@"./PublishedNodes/SimpleEvents.json")]
         public async Task CanEncodeWithReversibleEncodingTest(string publishedNodesFile) {
             // Arrange
             // Act
-            var result = await ProcessMessagesAsync(publishedNodesFile, arguments: new[] { "--mm=PubSub", "--UseReversibleEncoding=True" });
+            var result = await ProcessMessagesAsync(
+                publishedNodesFile,
+                new[] { "--mm=PubSub", "--UseReversibleEncoding=True" }
+            );
```
Comment (dacolgit) [inline modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/ReversibleEncodingIntegrationTests.cs]: add ConfigureAwait(false)?

File: modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/ReversibleEncodingIntegrationTests.cs [hunk 7]
```diff
@@ -107,13 +111,15 @@ public class ReversibleEncodingIntegrationTests : PublisherIntegrationTestBase,
...
+        [Theory]
         [InlineData(@"./PublishedNodes/SimpleEvents.json")]
         public async Task CanDecodeWithReversibleEncodingTest(string publishedNodesFile) {
             // Arrange
             // Act
-            var result = await ProcessMessagesAsync(publishedNodesFile, arguments: new[] { "--mm=PubSub", "--UseReversibleEncoding=True" });
+            var result = await ProcessMessagesAsync(
+                publishedNodesFile,
+                new[] { "--mm=PubSub", "--UseReversibleEncoding=True" }
+            );
```
Comment (dacolgit) [inline modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/ReversibleEncodingIntegrationTests.cs]: add ConfigureAwait(false)?

Comment (karok2m) [reply modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/ReversibleEncodingIntegrationTests.cs]: Added.

File: modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/ReversibleEncodingIntegrationTests.cs (hunk 6 above)
Comment (karok2m) [reply modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/ReversibleEncodingIntegrationTests.cs]: Added.

File: modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/ReversibleEncodingIntegrationTests.cs (hunk 5 above)
Comment (karok2m) [reply modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/ReversibleEncodingIntegrationTests.cs:35]: Added.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs (hunk 4 above)
Comment (karok2m) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs]: Ah, right, missed this one.

Comment (karok2m) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs]: Removed.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs (hunk 1 above)
Comment (karok2m) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs]: Done.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs (hunk 2 above)
Comment (karok2m) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs]: Done.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs (hunk 3 above)
Comment (karok2m) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs]: Done.

**Review by dacolgit — APPROVED** [review]

**Review by cristipogacean — APPROVED** [review]

**Pull request merged**